src/App.jsx

```diff
@@ -1,14 +1,71 @@
 import './App.css';
+import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
+import { useState } from 'react';
+
+// count like feature
+const calculateTotalLikes = (messages) =>
+  messages.reduce((total, message) => total + (message.liked ? 1 : 0), 0);
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [localColor, setLocalColor] = useState('color');
+  const [remoteColor, setRemoteColor] = useState('color');
+
+  // Determine local and remote participants dynamically
+  const localSender = messages.length > 0 ? messages[0].sender : 'Local';
+  const remoteSender = [...new Set(messages.map((msg) => msg.sender))]
+    .find((sender) => sender !== localSender) || 'Remote';
+
+  const totalLikes = calculateTotalLikes(messages);
+
+  const toggleLike = (id) => {
+    const updatedMessages = messages.map((msg) =>
+      msg.id === id ? { ...msg, liked: !msg.liked } : msg
+    );
+    setMessages(updatedMessages);
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>
+          Chat between{' '}
+          <span className={localColor}>{localSender}</span> and{' '}
+          <span className={remoteColor}>{remoteSender}</span>
```

Comment:
Note how these elements for the remote and local senders are essentially the same. You could argue that a component could be used for modularity and DRYness (basically having the same lines of code twice). Imagine if we needed to include all the members of a group chat?